### PR TITLE
Add changes to start the trx coordinator at the initiator side

### PR DESCRIPTION
**Pull request by lasinicl**
Fixes #<Issue Number>

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java [hunk 1]
```diff
@@ -4078,6 +4078,8 @@ private BLangOnFailClause createTrxInternalOnFail(Location pos, BLangSimpleVarRe
 
     @Override
     public void visit(BLangTransaction transactionNode) {
+        boolean previousInTransactionContext = this.inTransactionContext;
+        this.inTransactionContext = true;
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java:4082]: without checking whether we are in a trx scope here, shall we check it in CodeAnalyzer's ```visit(BLangInvocation.BLangActionInvocation actionInvocation)``` ? We have this check there already

Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java]: Fixed

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java [hunk 2]
```diff
@@ -575,7 +576,9 @@ public void visit(BLangTransaction transactionNode) {
         int previousCommitCount = this.commitCount;
         int previousRollbackCount = this.rollbackCount;
         boolean prevCommitRollbackAllowed = this.commitRollbackAllowed;
+        boolean prevWithinTransactionBlock = this.withinTransactionBlock;
```
Comment (gimantha) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java:579]: We already have a flag, called withinTransactionScope, I think we can use it

Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java]: Fixed

**Review by gimantha — APPROVED** [review]

**Pull request merged**